compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -304,6 +304,10 @@ private void defineConstructs(BLangPackage pkgNode, SymbolEnv pkgEnv) {
         pkgNode.typeDefinitions.forEach(typDef -> typDefs.add(typDef));
         defineTypeNodes(typDefs, pkgEnv);
 
```

Comment:
Shall we add a comment summarizing the reason for this.

Reply:
Added.
`Adding type info for global functions that are declared using var`
Would this comment be okay?

Reply:
How about specifying why we do this. Something like `Types of module level function pointers declared with 'var' are resolved before referencing them in functions as TypeNode is null for them.`

Reply:
Fixed. Thanks